src/useFind.ts

```diff
@@ -39,13 +39,19 @@ function loadServiceEventHandlers<
       return;
     }
 
+    let existing = false;
     data.value = data.value.map((item) => {
       if (getId(item) === getId(changedItem)) {
+        existing = true;
         return changedItem;
       }
 
       return item;
     });
+
+    if (!existing) {
+      data.value = [...data.value, changedItem];
+    }
```

Comment:
@mariusheine @anbraten What do you think of this instead?

```suggestion
    const itemIndex = data.value.findIndex((item) => getId(item) === getId(changedItem));
    if (itemIndex === -1) {
      data.value = [...data.value, changedItem];
    } else {
      data.value = [...data.value.slice(0, itemIndex), changedItem, ...data.value.slice(itemIndex + 1)];
    }
```

Reply:
@lukashass I really like it!

@anbraten ?

Reply:
LGTM. It keeps the order in case the item already exists.

Reply:
@anbraten @lukashass I applied the suggestion. Can one of you please approve then?